fondant/component_spec.py

```diff
@@ -300,58 +147,146 @@ def output_subsets(self) -> t.Mapping[str, ComponentSubset]:
         return types.MappingProxyType(
             {
                 name: ComponentSubset(subset)
-                for name, subset in self.fondant_component_specification[
-                    "output_subsets"
-                ].items()
+                for name, subset in self._specification["output_subsets"].items()
             }
         )
 
     @property
-    def input_arguments(self):
+    def accepts_additional_subsets(self) -> bool:
+        return self._specification["input_subsets"].get("additionalSubsets", True)
+
+    @property
+    def produces_additional_subsets(self) -> bool:
+        return self._specification["output_subsets"].get("additionalSubsets", True)
+
+    @property
+    def args(self) -> t.List[Argument]:
+        return [
+            Argument(
+                name=arg_name,
+                description=arg_info["description"],
+                type=arg_info["type"],
+            )
+            for arg_name, arg_info in self._specification["args"].items()
+        ]
+
+    @property
+    def kubeflow_specification(self) -> "KubeflowComponent":
+        return KubeflowComponent.from_fondant_component(self)
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}({self._specification!r}"
+
+
+class KubeflowComponent:
```

Comment:
```suggestion
class KubeflowSpec:
```
To be consistent with `FondantSpec` ;)